File: frame/map-frame-project/MapFrameCoreModule/src/main/java/com/grandtech/mapframe/core/sketch/ISketch.java
package com.grandtech.mapframe.core.sketch;

import androidx.annotation.NonNull;

import com.grandtech.mapframe.core.enumeration.GeometryType;
import com.mapbox.geojson.Geometry;

/**
 * @ClassName ISketch
 * @Description TODO 绘制需要实现的通用接口
 * @Author: fs
 * @Date: 2021/6/10 16:48
 * @Version 2.0
 */
public interface ISketch {


    /**
     * 设置绘制类型
     */
    public void setSketchType(@NonNull GeometryType geometryType);

    /**
     * 启动绘制
     */
    public void startSketch();


    public void stopSketch();

    /**
     * 判断当前绘制工具是否启动
     */
    public Boolean isStartSketch();


    /**
     * @Description 获取图形
     */
    public Geometry getSketchGeometry();


    /**
     * @Description 给绘制工具添加初始图形
     * @param geometry
     */
    public void addSketchGeometry(@NonNull Geometry geometry);
}

File: frame/map-frame-project/MapFrameCoreModuleV10/src/main/java/com/gradtech/mapframev10/core/sketch/ISketch.kt
package com.gradtech.mapframev10.core.sketch

import com.gradtech.mapframev10.core.enumeration.GeometryType
import com.mapbox.geojson.Geometry

/**
 * @ClassName ISketch
 * @Description TODO
 * @Author: fs
 * @Date: 2022/8/3 13:48
 * @Version 2.0
 */
interface ISketch {
    /**
     * 设置绘制类型
     */
    fun setSketchType(geometryType: GeometryType)

    /**
     * 启动绘制
     */
    fun startSketch()
    fun stopSketch()

    /**
     * 判断当前绘制工具是否启动
     */
    fun isStartSketch():Boolean

    /**
     * @Description 获取图形
     */
    fun getSketchGeometry(): SketchGeometry?

    /**
     * @Description 给绘制工具添加初始图形
     * @param geometry
     */
    fun setSketchGeometry(geometry: Geometry)

    fun clearSketch()

    fun tryUnDo()

    fun tryReDo()
}